Kudu.Core/Infrastructure/FileSystemHelpers.cs

```diff
@@ -290,6 +289,27 @@ public static void DeleteDirectoryContentsSafe(string path, bool ignoreErrors =
             DeleteDirectoryContentsSafe(Instance.DirectoryInfo.FromDirectoryName(path), ignoreErrors);
         }
 
+        public static bool IsFileSystemReadOnly(IEnvironment environment)
+        {
+            try
+            {
+                // Temp folder will be clean up everytime site starts up
+                // So it is fine to have leftover folder if deletion failed
+                string folderName = Path.Combine(environment.DataPath, "Temp", Guid.NewGuid().ToString("N"));
+                Instance.Directory.CreateDirectory(folderName);
+                Instance.Directory.Delete(folderName);
+                return false;
```

Comment:
we will still need a expiration time if FS is in RW mode, so that we do not introduce too many I/O.
working on the improvement.